### PR TITLE
fix(runner): setting up env instead in scheduler

### PR DESCRIPTION
**Pull request by bojiang**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

Feel free to tag members/contributors who can help review your PR.

### CODE REVIEW

File: bentoml/_internal/runner/strategy.py [hunk 1]
```diff
@@ -88,42 +96,55 @@ def get_worker_count(
...
         cls,
         runnable_class: t.Type[Runnable],
         resource_request: dict[str, t.Any] | None,
         worker_index: int,
-    ) -> None:
+    ) -> dict[str, str]:
         if resource_request is None:
             resource_request = system_resources()
 
+        env: dict[str, str] = {}
```
Comment (aarnphm) [inline bentoml/_internal/runner/strategy.py:108]:
```suggestion
        env: dict[str, str] = os.environ.copy()
```

File: bentoml/_internal/server/cli/api_server.py [hunk 2]
```diff
@@ -69,6 +70,10 @@ def main(
     \b
     This is an internal API, users should not use this directly. Instead use `bentoml serve <path> [--options]`
     """
+    import logging
+
+    logger = logging.getLogger(__name__)
+    logger.info(f"Starting BentoML API server {worker_id}...")
```
Comment (aarnphm) [inline bentoml/_internal/server/cli/api_server.py:76]:
```suggestion
    logger.debug(f"Starting BentoML API server {worker_id}...")
```
Should be a debug message.

File: bentoml/_internal/server/cli/runner.py [hunk 3]
```diff
@@ -1,3 +1,8 @@
+"""
+All entrypoints script should keep the requirements(imports) simple.
+We shall avoid import anything from bentoml here.
+"""
+
```
Comment (aarnphm) [inline bentoml/_internal/server/cli/runner.py:5]: Remove this. We can mention this in a README.md for this directory.

File: bentoml/_internal/server/cli/runner.py [hunk 4]
```diff
@@ -91,6 +91,10 @@ def main(
         arbiter.start()
         return
 
+    from bentoml import load
+
+    from ...context import component_context
```
Comment (aarnphm) [inline bentoml/_internal/server/cli/runner.py:96]: Let's do absolute imports here. We can run this script standalone.

**Comment by bojiang** [comment]
I will reopen this PR to the grpc branch, since at that branch we have big changes about entrypoint codes

**Pull request closed without merging**